mongox/models.py

```diff
@@ -114,6 +114,25 @@ async def get(self) -> T:
             raise MultipleMatchesFound()
         return objects[0]
 
+    async def get_or_create(self, defaults: typing.Dict = dict()) -> T:
+        """
+        Get the only document matching or created the document
+        """
+
+        objects = await self.limit(2).all()
+        if len(objects) == 2:
+            raise MultipleMatchesFound()
+        if len(objects) == 0:
+            data = {expression.key: expression.value for expression in self._filter}
+
+            for key, value in defaults.items():
+                key_ = key if isinstance(key, str) else key._name
+                data[key_] = value
+
+            result = await self._collection.insert_one(data)
```

Comment:
And use the existing insert here?

Reply:
You are right, i could use the same insert method already existing.